### DISCUSSION TITLE
Introducing the PyRI Open Source Teach Pendant

### DISCUSSION THREAD

**Discussion by johnwason** [pyodide/pyodide#1803] [General]
https://github.com/pyri-project/pyri-core/blob/master/README.md

The Python Restricted Industrial (PyRI) Open Source Teach Pendant is a high-level programming interface for robotics and automation systems. It is intended to be using with Robot Raconteur and ROS projects. The user interface is designed to run in a web browser, so that any laptop or computer can be used to access the user interface without needing to install software. Pyodide has been extremely useful for this project, since Python is used extensively in robotics. We would have had to port a lot of code to JavaScript or Typescript if Pyodide wasn't available. The gory details of how everything works is explained here: https://github.com/pyri-project/pyri-core/blob/master/doc/pyri_software_architecture.md

This project used a fork of Pyodide called Robot Raconteur Pyodide (https://github.com/robotraconteur/robotraconteur_pyodide). I have had a really difficult time getting the C++ Robot Raconteur module to work in the stock Pyodide build. I have had to statically link the Robot Raconteur native module into the Python build. I am hoping this is something that can be resolved over time as both projects progress.

**Comment by hoodmane** [comment]
If you like, we could add this to [related projects](https://pyodide.org/en/stable/project/related-projects.html).
